pkg/keystone/deployment.go

```diff
@@ -37,11 +38,43 @@ func Deployment(
 ) *appsv1.Deployment {
 	runAsUser := int64(0)
 
+	livenessProbe := &corev1.Probe{
+		// TODO might need tuning
+		PeriodSeconds:       3,
+		InitialDelaySeconds: 3,
+	}
+	readinessProbe := &corev1.Probe{
+		// TODO might need tuning
+		PeriodSeconds:       5,
+		InitialDelaySeconds: 5,
+	}
+
 	args := []string{"-c"}
 	if instance.Spec.Debug.Service {
 		args = append(args, DebugCommand)
+		livenessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/bin/true",
+			},
+		}
+
+		readinessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/bin/true",
+			},
+		}
 	} else {
 		args = append(args, ServiceCommand)
+
+		//
+		// https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+		//
+		livenessProbe.HTTPGet = &corev1.HTTPGetAction{
+			Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(KeystonePublicPort)},
+		}
+		readinessProbe.HTTPGet = &corev1.HTTPGetAction{
+			Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(KeystonePublicPort)},
+		}
```

Comment:
Is the default `/` path okay for these?  I'm guessing it is

Reply:
yes, see  [1] , which redirects to the admin api [2] and / is to "Discover API version information". The check could probably improved, but should work for an initial start.

[1] https://docs.openstack.org/keystone/pike/api_curl_examples.html#get
[2] https://docs.openstack.org/keystone/pike/api_curl_examples.html#id1